src/ChunkingContext.jl

```diff
@@ -0,0 +1,82 @@
+# When splitting the work among multiple tasks, each task should have at least this many bytes of input
+# This is to avoid having too many tasks with too little work to do.
+# TODO: make this configurable and find a good default (the current 16 KiB is a guess)
+const MIN_TASK_SIZE_IN_BYTES = 16 * 1024
+
+_comment_to_bytes(x::AbstractString) = Vector{UInt8}(x)
+_comment_to_bytes(x::Char) = _comment_to_bytes(ncodeunits(x) > 1 ? string(x) : UInt8(x))
+_comment_to_bytes(x::UInt8) = [x]
+_comment_to_bytes(x::Vector{UInt8}) = x
+_comment_to_bytes(::Nothing) = nothing
+
+# Holds a byte buffer and newline positions for a single chunk of the input file.
+# The newline positions are used to split the chunk into tasks for parallel parsing.
+struct ChunkingContext
```

Comment:
This is part of the public API of the package - users are meant to construct one of these.

Therefore, this should have a docstring, which should contain the signature of its constructor and how to use it.
You can put that docstring either on this struct or on the outer constructor, up to you. :)

Reply:
(this pattern should be followed throughout the package: public APIs should have docstrings so that users can use `help` to learn how to use your package.)